packages/lending/src/adapter.ts

```diff
@@ -204,13 +204,26 @@ export class Adapter extends common.Web3Toolkit {
     return await protocol.getPortfolio(account, marketId);
   }
 
+  /**
+   * Retrieves an array of protocol information (tokens) from all registered protocols.
+   *
+   * @returns {Promise<ProtocolInfo[]>} an array of protocol information (tokens)
+   */
+  async getProtocolInfos(): Promise<ProtocolInfo[]> {
+    const protocolInfos = await Promise.all(
```

Comment:
wondering if we need to await here since we return a Promise?

Reply:
We need to use `flatten` after this to ensure all `getProtocolInfos` calls are complete. The await ensures that all these calls finish before we combine their results with `flatten`.